File: testdata/fixtures/javascript/CWE-798.js
// CWE-798: Hardcoded credential - secret literal in source.
const config = require('config');

// ruleid: js-hardcoded-credential
const password = 'hunter2-long-secret-123';
// ruleid: js-hardcoded-credential
const api_key = 'test-api-key-1234567890';
// ruleid: js-hardcoded-credential
const secret = 's3cr3t-value-123';
// ruleid: js-hardcoded-credential
const token = 'ghp_1234567890abcdef1234567890abcdef';

function safeEnv() {
  // ok: js-hardcoded-credential
  const password = process.env.PASSWORD;
  // ok: js-hardcoded-credential
  const apiKey = process.env.API_KEY;
  // ok: js-hardcoded-credential
  const secret = getSecret();
  // ok: js-hardcoded-credential
  const token = '';
}

File: testdata/fixtures/python/CWE-798.py
# CWE-798: Hardcoded credential — a secret literal is written into the source.
#
# A password or API key written as a string literal is in every checkout, every
# binary and every log. It cannot be rotated without a release. Read it from
# the environment or a vault instead, or leave the placeholder empty.
import os

# ruleid: python-hardcoded-credential
PASSWORD = "hunter2"
# ruleid: python-hardcoded-credential
api_key = "AKIAIOSFODNN7EXAMPLE"
# ruleid: python-hardcoded-credential
secret = "s3cr3t-value-123"
# ruleid: python-hardcoded-credential
TOKEN = "ghp_1234567890abcdef"
# ruleid: python-hardcoded-credential
pwd = "my-password-123"


def with_env():
    # ok: python-hardcoded-credential
    password = os.environ.get("PASSWORD")
    # ok: python-hardcoded-credential
    api_key = os.getenv("API_KEY")
    # ok: python-hardcoded-credential
    secret = get_secret()
    # ok: python-hardcoded-credential
    token = ""
    # ok: python-hardcoded-credential
    pwd = None
